CMakeLists.txt

```
@@ -16,13 +16,13 @@ if(NOT CMAKE_BUILD_TYPE)
   if (EXISTS "${CMAKE_SOURCE_DIR}/.git")
     message(WARNING "CMAKE_BUILD_TYPE not specified, assuming CMAKE_BUILD_TYPE=Debug because .git exists.")
     set(default_build_type "Debug")
+    set(CMAKE_BUILD_TYPE "${default_build_type}" CACHE
+        STRING "Default BUILD_TYPE is Debug, other options are: RelWithDebInfo, Release, and MinSizeRel." FORCE)
   else()
-    message(WARNING "CMAKE_BUILD_TYPE not specified, setting to RelWithDebInfo because .git does NOT exist.")
-    set(default_build_type "RelWithDebInfo")
+    message(WARNING "CMAKE_BUILD_TYPE not specified, leaving unset because .git does NOT exist.")
```

Since we seem to have folks continually come in thinking this is **wrong**, can we add a comment like:

```
```suggestion
    # Hello, you may be thinking about the wisdom of the leaving CMAKE_BUILD_TYPE unset. Consider the known ramifications of changing it:
    #  * RelWithDebInfo/Release/MinSizeRel will disable asserts.
    #  * Debug has performance drawbacks for general QA.
    #  ....
    message(WARNING "CMAKE_BUILD_TYPE not specified, leaving unset because .git does NOT exist.")
```

---

Why isn't it wrong? I think that we relied on the fact that it worked, but I'm not sure it's well defined.

---

Sorry, I'm not saying that we shouldn't perhaps change it but there are gotchas with the uninformed change of the default (unset). If those gotchas are undesirable, maybe you shouldn't do it. It's not the first time this was discussed...

---

For now it was agreed to leave the fallback to reldeb. If we decide to go back to "unset" type -  I agree we should add proper info regarding it.